packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/test/Providers/TypeProviderTests.cs

```diff
@@ -223,18 +223,23 @@ public async Task TestCanCustomizeTypeWithChangedNamespace()
         {
             await MockHelpers.LoadMockGeneratorAsync(compilation: async () => await Helpers.GetCompilationFromDirectoryAsync());
             var testTypeProvider = new TestTypeProvider();
+            Assert.IsNull(testTypeProvider.CustomCodeView);
 
-
+            testTypeProvider.Update(name: "Foo"); // custom code only take effect when the name matches
```

Comment:
So this update is here to mimic as if there were a visitor updating this?
If so why not we just write a visitor here

Reply:
Yes, it is to mimic [NamespaceVisitor](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/packages/http-client-csharp/generator/Azure.Generator/src/Visitors/NamespaceVisitor.cs)
They are basically the same thing, the visitor just call the update method. I am fine with both.